cfme/tests/storage/test_volume.py

```diff
@@ -46,7 +51,48 @@ def volume(appliance, provider):
             msg=str(e)))
 
 
-def test_storage_volume_create_cancelled_validation(appliance, provider):
+@pytest.fixture(scope="function")
+def instance_fixture(appliance, provider, small_template):
+    instance = appliance.collections.cloud_instances.instantiate(random_vm_name('stor'),
+                                                                 provider,
+                                                                 small_template.name)
+    if not instance.exists_on_provider:
+        instance.create_on_provider(allow_skip="default", find_in_cfme=True)
+    elif instance.provider.one_of(EC2Provider) and instance.mgmt.state == VmState.DELETED:
+        instance.mgmt.rename('test_terminated_{}'.format(fauxfactory.gen_alphanumeric(8)))
+        instance.create_on_provider(allow_skip="default", find_in_cfme=True)
+    yield instance
+    instance.cleanup_on_provider()
+
+
+def create_volume(appliance, provider, is_from_manager=False, az=None, cancel=False,
+                  should_assert=False):
+    volume_collection = appliance.collections.volumes
+    manager = appliance.collections.block_managers.filter({"provider": provider}).all()[0]
+    if provider.one_of(OpenStackProvider):
+        volume = volume_collection.create(name=fauxfactory.gen_alpha(),
+                                          storage_manager=manager,
+                                          tenant=provider.data['provisioning']['cloud_tenant'],
+                                          volume_size=STORAGE_SIZE,
+                                          provider=provider,
+                                          cancel=cancel,
+                                          from_manager=is_from_manager)
+    elif provider.one_of(EC2Provider):
+        az = az if az else provider.region + 'a'
```

Comment:
Optional:
I would avoid to use string addition; we can handle with `.format`

Reply:
Sure, will rework with format